autometa/taxonomy/get_lca_taxa.py

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env python
+
+import os
+import glob
+import logging
+import pyreadr
+
+import pandas as pd
+import numpy as np
+
+from autometa.taxonomy.lca import LCA
+from autometa.taxonomy.ncbi import NCBI, NCBI_DIR
+
+logger = logging.getLogger(__name__)
+
+# TODO: Currently hardcoded. Need to change it. Not sure how.
+
+lca = LCA(dbdir="/media/bigdrive1/Databases/autometa_databases", verbose=True)
+ncbi = NCBI(dirpath="/media/bigdrive1/Databases/autometa_databases", verbose=True)
+
+
+def map_taxid(input_fpath, cog_org_df, output_fpath):
+    """Find the lca of the query and subject seq ID, map it's taxid and taxonomic rank
+
+    Parameters
+    ----------
+    input_fpath : str
+        Path to input dataframe
+    cog_org_df : DataFrame
+        cog-org-20.csv downloaded from the FTP. This maps the assembly accession to the taxid
+    output_fpath : str
+        Path to output directory
+
+    """
+
+    # Loading the RDS file in python
+    rds = pyreadr.read_r(input_fpath)
+    # extracting the df from the dictionary. It only has one key ie. None
+    df = rds[None]
+    # removing version numbers
+    df["sseqid"] = df["sseqid"].str.split(".").str[0]
+    df["qseqid"] = df["qseqid"].str.split(".").str[0]
+    # Creating new column with NA as default values
+    df["sseqid_taxid"] = "NA"
+    df["qseqid_taxid"] = "NA"
+
+    df_dict = dict(list(df.groupby("qseqid_assembly")))
+    for accession, value in df_dict.items():
+        taxid = cog_org_df.loc[accession, "NCBI_taxID"]
+        # See https://stackoverflow.com/a/27045135/12671809
+        df.loc[df["qseqid_assembly"] == accession, "qseqid_taxid"] = taxid
+        # Since we are doing an all-vs-all blast each sseqid would also be a qseqid so we don't have to build teh above dictionary again
+        df.loc[df["sseqid_assembly"] == accession, "sseqid_taxid"] = taxid
+
+    df["lca_taxid_ncbi"] = "NA"
+    df["lca_rank_ncbi"] = "NA"
+
+    # See https://www.kite.com/python/answers/how-to-apply-a-function-to-two-columns-in-a-pandas-dataframe-in-python#:~:text=Use%20any%20two%2Dinput%20function%20to%20obtain%20a%20new%20column,function%20which%20takes%20two%20inputs
+    # See https://stackoverflow.com/a/49192682/12671809
+    df[["lca_taxid_ncbi", "lca_rank_ncbi"]] = df[
+        ["qseqid_taxid", "sseqid_taxid"]
+    ].apply(get_lca, axis=1, result_type="expand")
+    df_rds_name = os.path.basename(input_fpath)
+    df_fname = (os.path.splitext(df_rds_name)[0]) + ".tsv"
+    df.to_csv(os.path.join(output_fpath, df_fname), sep="\t", header=True, index=False)
+    logger.info(f"Wrote {df_fname}.")
```

Comment:
Minor nitpicks here

### First suggestion..

```suggestion
    outfilename = os.path.splitext(os.path.basename(input_fpath))[0] + ".tsv"
    out = os.path.join(outdir, df_fname)
    df.to_csv(out, sep="\t", header=True, index=False)
    logger.info(f"Wrote {df_fname}.")
```

### Second suggestion.. probably the way to go here?

```suggestion
    df.to_csv(out, sep="\t", header=True, index=False)
    logger.info(f"Wrote {out}.")
```